### PR TITLE
Add notebook verification for visualization examples.

### LINKED ISSUE #683: [RFC] CI for examples/visualization
It'll be better if we can validate that all the visualization examples run without failures with CI.

I guess https://github.com/computationalmodelling/nbval/ might be useful.

### PR DESCRIPTION
**Pull request by ytsmiling**
This PR makes CI checks whether notebooks run without failure. This PR resolves #683.

### CODE REVIEW

File: .github/workflows/examples.yml [hunk 1]
```diff
@@ -32,6 +32,9 @@ jobs:
 
         # Install all dependencies needed for examples.
         pip install --progress-bar off $(ls dist/*.tar.gz)[example]
+
+        # Install additional test-specific dependencies
+        pip install --progress-bar off nbval
```
Comment (ytsmiling) [inline .github/workflows/examples.yml:37]: Is it okay to add test-specific dependencies here, or shall I move to examples' requirements?

File: .github/workflows/examples.yml [hunk 2]
```diff
@@ -51,6 +54,12 @@ jobs:
             python $file --pruning > /dev/null
           fi
         done
+
+        for file in `find examples -name '*.ipynb'`
```
Comment (ytsmiling) [inline .github/workflows/examples.yml:58]: Should I split this test case like `Run multi-node examples`?

File: .github/workflows/examples.yml (hunk 1 above)
Comment (sile) [reply .github/workflows/examples.yml:37]: I think that it's not a bad idea to install `nbval` here, but for future maintainability, it seems better to add it to the "example" dependency list defined in [setup.py](https://github.com/optuna/optuna/blob/master/setup.py#L57).

File: .github/workflows/examples.yml (hunk 2 above)
Comment (sile) [reply .github/workflows/examples.yml:58]: I prefer to split.

File: examples/visualization/plot_study.ipynb [hunk 3]
```diff
@@ -136,7 +136,7 @@
         "def objective(trial):\n",
         "    \n",
         "    clf = MLPClassifier(\n",
-        "        hidden_layer_sizes=tuple([trial.suggest_int(f'n_units_l{i}', 32, 64) for i in range(3)]),\n",
+        "        hidden_layer_sizes=tuple([trial.suggest_int('n_units_l{}'.format(i), 32, 64) for i in range(3)]),\n",
```
Comment (sile) [inline examples/visualization/plot_study.ipynb:139]: Nice catch 👍

File: setup.py [hunk 4]
```diff
@@ -61,6 +61,7 @@ def get_extras_require() -> Dict[str, List[str]]:
             "mlflow",
             "mpi4py",
             "mxnet",
+            "nbeval",
```
Comment (crcrpar) [inline setup.py:64]: Isn't this a typo? 

ref: https://github.com/computationalmodelling/nbval#installation

```suggestion
            "nbval",
```

Comment (ytsmiling) [reply setup.py:64]: Thank you!

Comment (ytsmiling) [reply setup.py:64]: It is! Thank you!

**Pull request merged**